### PR TITLE
Discord Provider

### PR DESCRIPTION
**Pull request by l1n**
Implemented per https://discordapp.com/developers/docs/topics/oauth2.

### CODE REVIEW

**Comment by ploxiln** [comment]
I'd rather the import path not be renamed from `github.com/bitly/oauth2_proxy` to github.com/ploxiln/oauth2_proxy` in this PR. It may be a bit awkward, but you can clone my repo to `$GOPATH/src/github.com/bitly/oauth2_proxy`, or if you already have the bitly repo there you can add my repo as another remote and check out my master branch, etc. I'll probably do the renaming soon - but separately.

Other than that, there's just some minor whitespace/formatting issues.

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ to validate accounts by email, domain or group.
 ## This Fork
 
 This repo [ploxiln/oauth2_proxy](https://github.com/ploxiln/oauth2_proxy)
-is a fork of [bitly/oauth2_proxy](https://github.com/bitly/oauth2_proxy)
+is a fork of [ploxiln/oauth2_proxy](https://github.com/ploxiln/oauth2_proxy)
```
Comment (ploxiln) [inline README.md]: ... and you were a bit overzealous with the global search and replace :)

**Comment by l1n** [comment]
Fair enough, reverting that commit 👍 I didn't mean to add it in this pull request since it wasn't part of the Discord provider.

**Comment by ploxiln** [comment]
Thanks!

(I pushed up one last gofmt tweak ... the trick is `gofmt -w providers/*.go`)

**Comment by l1n** [comment]
I apologize for not speaking Go too natively, my last Go project was a year or so ago 😅 thank you for your help

File: providers/discord.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		p.ProfileURL = &url.URL{Scheme: "https",
+			Host: "discordapp.com",
+			Path: "/api/users/@me",
+		}
+	}
+	if p.ValidateURL.String() == "" {
+		p.ValidateURL = p.ProfileURL
+	}
+	if p.Scope == "" {
+		p.Scope = "identify email connections"
```
Comment (ploxiln) [inline providers/discord.go:53]: looks like "connections" is not needed

Comment (ploxiln) [reply providers/discord.go:53]: this and some other little refinements could be in a follow-up pull request if someone cares enough :)

(e.g. removing parts of DiscordUserInfo which are unused, maybe checking Verified property ...)

**Comment by l1n** [comment]
It isn't needed right now, but I'd like to extend from the user and email that are by default provided to the proxied host at some point: similar to the way that `mod_shib` provides `displayName` https://tcg.stanford.edu/?p=131

**Pull request merged**